strawberryfields/engine.py

```diff
@@ -336,8 +360,144 @@ def run(self, program, *, shots=1, compile_options={}, modes=None, state_options
             # empty sequence
             pass
         else:
-            result.state = self.backend.state(modes, **state_options)  # tfbackend.state can use kwargs
+            result.state = self.backend.state(
+                modes, **state_options
+            )  # tfbackend.state can use kwargs
         return result
 
 
+class StarshipEngine(BaseEngine):
+    """
+    Starship quantum program executor engine.
+
+    Executes :class:`.Program` instances on the chosen remote backend, and makes
+    the results available via :class:`.Result`.
+
+    Args:
+        backend (str, BaseBackend): name of the backend, or a pre-constructed backend instance
+    """
+
+    API_DEFAULT_REFRESH_SECONDS = 1
+    HARDWARE_BACKENDS = ("chip0",)
+
+    def __init__(self, api_client_params=None):
+        # Only chip0 backend supported initially.
+        backend = "chip0"
+        super().__init__(backend)
+
+        api_client_params = api_client_params or {}
+        self.client = APIClient(**api_client_params)
+        self.jobs = []
+
+    def __str__(self):
+        return self.__class__.__name__ + "({})".format(self.backend_name)
+
+    def reset(self, backend_options=None):
+        """
+        Reset must be called in order to submit a new job. This clears the job queue as well as
+        any ran Programs.
+        """
+        if backend_options is None:
+            backend_options = {}
+
+        super().reset(backend_options)
+        self.jobs.clear()
+
+    def _init_backend(self, *args):
+        """
+        TODO: This does not do anything right now.
+        """
+        # Do nothing for now...
+        pass
```

Comment:
Yes

Reply:
Strange. I was going to say, it doesn't seem needed for a remote backend, but perhaps this will be the place where https://github.com/XanaduAI/strawberryfields/pull/101#discussion_r300222618 goes.

Reply:
The method could be removed, it's just a one-liner in LocalEngine.

Reply:
@smite I wonder if the call to `_init_backend` in `_run` should be moved out of the for loop (line 278 in `engine.py`). I don't think it makes sense to remove this method from the abstract base class until we refactor how it is used in `_run`. Thoughts?

Reply:
Yeah, that's probably a good idea. But most likely `StarshipEngine.run()` should not call `BaseEngine._run()` at all, since what it needs to do is so different from what a local engine needs to do.

It seems to me StarshipEngine (at least for now) only needs to be able to run a single Program at a time, so the loop isn't necessary. No backend init is needed on the client side. 
All it needs from `BaseEngine._run()` is the Program compilation/locking part and the assignment of `self.samples`, so it's probably easiest to add these directly to `StarshipEngine.run()`.